Makefile

```diff
@@ -59,12 +59,15 @@ rule_file = $(notdir $(1))
 rule_path = $(patsubst %/,%,$(dir $(1)))
 last_path = $(notdir $(patsubst %/,%,$(dir $(1))))
 ansicolor = $(shell echo $(call last_path,$(1)) | cksum | cut -b1-2 | xargs -IS expr 2 \* S + 17)
-CPRINTF = @if [ -t 1 ]; then \
+emacs_out = @printf "  %10s %s/%s\n" $(1) $(call rule_path,$(2)) $(call rule_file,$(2))
```

Comment:
But this is the same as the simpler alternative that doesn't split and rejoin the paths. Why the duplication?

Also, Emacs isn't the only circumstance in which a terminal doesn't support 256 colors. You could use `tput colors` and check for 256. (It returns -1 in Emacs dumb terminal.)

Reply:
Emacs compilation window sets the TERM to 'dumb': if we hit this exactly the emacs_out is used, otherwise we fold back to the color_printf that has the TTY check. I can add the 256 color check for that one if you want.

Reply:
Point is that we don't need to special case Emacs if we just check for 256 colors.

And as mentioned above, why use this more complicated format string instead of `"  %10s %s\n"` which produces the same output with simpler code and already exists (in the else clause).

Sorry if this seems pedantic, but this code is sure to be revisited and any unnecessary complexity will make that slower and more error-prone. When doing maintenance, especially on features relating to portability, the programmer will assume that any extra conditional is actually essential for some environment until proven otherwise.

I think we can all admit that this feature is a bit of a gimmick. I don't know why we're spending time on it right now when there are really important things to be done. I can just merge it as is and we can fix/improve it when we get user support issues in the future. I find that a bit unsatisfying and unprofessional, but we really need to focus on the important stuff.